lighthouse-core/gather/driver.js

```diff
@@ -1581,6 +1586,149 @@ class Driver {
 
     await this.sendCommand('Page.enable');
   }
+
+  /**
+   * The Fetch domain accepts patterns for controlling what requests are intercepted, but we
+   * enable the domain for all patterns and filter events at a lower level to support multiple
+   * concurrent usages. Reasons for this:
+   *
+   * 1) only one set of patterns may be applied for the entire domain.
+   * 2) every request that matches the patterns are paused and only resumes when certain Fetch
+   *    commands are sent. So a listener of the `Fetch.requestPaused` event must either handle
+   *    the requests it cares about, or explicitly allow them to continue.
+   * 3) if multiple commands to continue the same request are sent, protocol errors occur.
+   *
+   * So instead we have one global `Fetch.enable` / `Fetch.requestPaused` pair, and allow specific
+   * urls to be intercepted via `driver.setOnRequestPausedHandler`.
+   */
+  async enableRequestInterception() {
+    await this.sendCommand('Fetch.enable', {
+      patterns: [{requestStage: 'Request'}, {requestStage: 'Response'}],
+    });
+    await this.on('Fetch.requestPaused', this._onRequestPaused);
+  }
+
+  /**
+   * @param {string} url
+   * @param {(event: LH.Crdp.Fetch.RequestPausedEvent) => void} handler
+   */
+  async setOnRequestPausedHandler(url, handler) {
+    this._onRequestPausedHandlers.set(url, handler);
+  }
+
+  /**
+   * @param {LH.Crdp.Fetch.RequestPausedEvent} event
+   */
+  async _onRequestPaused(event) {
+    const handler = this._onRequestPausedHandlers.get(event.request.url);
+    if (handler) {
+      await handler(event);
+    } else {
+      // Nothing cares about this URL, so continue.
+      await this.sendCommand('Fetch.continueRequest', {requestId: event.requestId});
+    }
+  }
+
+  async disableRequestInterception() {
+    await this.sendCommand('Fetch.disable');
+    await this.off('Fetch.requestPaused', this._onRequestPaused);
+    this._onRequestPausedHandlers.clear();
+  }
+
+  /**
+   * Requires that `driver.enableRequestInterception` has been called.
+   *
+   * Fetches any resource in a way that circumvents CORS.
+   *
+   * @param {string} url
+   * @param {number} timeoutInMs
+   */
+  async fetchArbitraryResource(url, timeoutInMs = 500) {
+    if (!this.isDomainEnabled('Fetch')) {
+      throw new Error('Fetch domain must be enabled to use fetchArbitraryResource');
+    }
+
+    /** @type {Promise<string>} */
+    const requestInterceptionPromise = new Promise((resolve, reject) => {
+      this.setOnRequestPausedHandler(url, async (event) => {
+        const {requestId, responseStatusCode} = event;
+
+        // The first requestPaused event is for the request stage. Continue it.
+        if (!responseStatusCode) {
+          // Remove same-site cookies so we aren't buying stuff on Amazon.
```

Comment:
I guess we're not concerned with the sort of manifest/user profile state being stored in a samesite cookie that made us want to deal with cookies at all in the first place?

Reply:
idk how cookies work tbh. so you're saying this bit of code totally invalidates any source maps behind cookie-authentication?

Reply:
any cookie auth that they've marked as SameSite, yeah

marking auth cookies as samesite is a decent way to prevent the [annoying site-style logout/language attack](https://github.com/feross/TheAnnoyingSite.com)